pkg/operator/resource/resourceapply/apiregistration.go

```diff
@@ -27,6 +27,14 @@ func ApplyAPIService(client apiregistrationv1client.APIServicesGetter, recorder
 	modified := resourcemerge.BoolPtr(false)
 	existingCopy := existing.DeepCopy()
 
+	// Since the required object does not specify a service port, and
+	// the port in the existing object is likely to be defaulted to
+	// 443, avoid thrashing by ensuring that the port is not
+	// considered when determining equality.
+	if required.Spec.Service != nil && existingCopy.Spec.Service != nil {
+		required.Spec.Service.Port = existingCopy.Spec.Service.Port
```

Comment:
this should rather set the default value 443.

Note that it is not good to mutate an incoming object `required`.

Reply:
Ok, no need for this PR then.